test/plugins/FakeTimeStampDataGenerator.cpp

```diff
@@ -0,0 +1,190 @@
+  TLOG_DEBUG(TLVL_ENTER_EXIT_METHODS) << get_name() << ": Entering do_stop() method";
+  thread_.stop_working_thread();
+  TLOG() << get_name() << " successfully stopped";
+  TLOG_DEBUG(TLVL_ENTER_EXIT_METHODS) << get_name() << ": Exiting do_stop() method";
+}
+
+void
+FakeTimeStampDataGenerator::do_scrap(const nlohmann::json& /*args*/)
+{
+  TLOG_DEBUG(TLVL_ENTER_EXIT_METHODS) << get_name() << ": Entering do_scrap() method";
+  TLOG_DEBUG(TLVL_ENTER_EXIT_METHODS) << get_name() << ": Exiting do_scrap() method";
+}
+
+ std::vector<triggeralgs::TimeStampedData>
+FakeTimeStampDataGenerator::GetTimestamp()
+{
+  std::vector<triggeralgs::TimeStampedData> tsds;
+  triggeralgs::TimeStampedData tsd{};
+
+  int signaltype = rdm_signaltype(generator);
+
+  auto tsd_start_time = std::chrono::steady_clock::now();
+  tsd.time_stamp = (uint64_t)pd_clock(tsd_start_time.time_since_epoch()).count();
+  tsd.signal_type = signaltype;
+  auto now = std::chrono::steady_clock::now();
+  tsd.counter = (uint32_t)pd_clock(now.time_since_epoch()).count();
+
+  //std::cout << "\033[32mtsd.timestamp: " << tsd.time_stamp << "\033[0m  ";
+  std::cout << "\033[32m" << tsd.time_stamp << ", "<< tsd.signal_type << ", "<< tsd.counter << "\033[0m\n";
+
+  tsds.push_back(tsd);
+  
+  return tsds;
+}
+
+void
+FakeTimeStampDataGenerator::do_work(std::atomic<bool>& running_flag)
+{
+  TLOG_DEBUG(TLVL_ENTER_EXIT_METHODS) << get_name() << ": Entering do_work() method";
+  size_t generatedCount = 0;
+  size_t sentCount = 0;
+
+  while (running_flag.load()) 
+  {
+    TLOG_DEBUG(TLVL_GENERATION) << get_name() << ": Start of sleep between sends";
+    std::this_thread::sleep_for(std::chrono::nanoseconds(1000000000));
+
+    std::vector<triggeralgs::TimeStampedData> tsds = GetTimestamp();
+
+    if (tsds.size() == 0) 
+    {
+      std::ostringstream oss_prog;
+      oss_prog << "Last TSDs packet has size 0, continuing!";
+      ers::debug(dunedaq::dunetrigger::ProgressUpdate(ERS_HERE, get_name(), oss_prog.str()));
+      continue; 
+    } 
+    else 
+    {
+      std::ostringstream oss_prog;
+      ers::debug(dunedaq::dunetrigger::ProgressUpdate(ERS_HERE, get_name(), oss_prog.str()));
+    }
+
+    generatedCount+=tsds.size();
+
+    std::string thisQueueName = outputQueue_->get_name();
+    bool successfullyWasSent = false;
+    while (!successfullyWasSent && running_flag.load())
+    {
+      TLOG_DEBUG(TLVL_GENERATION) << get_name() << ": Pushing the generated TSD onto queue " << thisQueueName;
+      for (auto const& tsd: tsds)
+      {
+	try
+	{
+	  outputQueue_->push(tsd, queueTimeout_);
+	  successfullyWasSent = true;
+	  ++sentCount;
+	}
+	catch (const dunedaq::appfwk::QueueTimeoutExpired& excpt)
+	{
+	  std::ostringstream oss_warn;
+	  oss_warn << "push to output queue \"" << thisQueueName << "\"";
+	  ers::warning(dunedaq::appfwk::QueueTimeoutExpired(ERS_HERE, get_name(), oss_warn.str(),
+							    std::chrono::duration_cast<std::chrono::milliseconds>(queueTimeout_).count()));
```

Comment:
`queueTimeout_` is already a `std::chrono::milliseconds`, so the cast is unnecessary